internal/query/event.go

```diff
@@ -66,7 +66,13 @@ func filterAuditLogRetention(ctx context.Context, auditLogRetention time.Duratio
 	if callTime.IsZero() {
 		callTime = time.Now()
 	}
-	return builder.CreationDateAfter(callTime.Add(-auditLogRetention))
+	oldestAllowed := callTime.Add(-auditLogRetention)
+	// The audit log retention time should overwrite the creation date after query only if it is older
+	// For example API calls should still be able to restrict the creation date after to a more recent date
+	if builder.GetCreationDateAfter().Before(oldestAllowed) {
+		return builder.CreationDateAfter(oldestAllowed)
+	}
```

Comment:
don't we need to also check the `CreationDateBefore()`?
resp. if no creationDateAfter is specified we still need to make sure you can't go past the retention

if somehow possible, please add some tests

Reply:
ok just noticed, that if not set the `builder.GetCreationDateAfter()` will obviously return a zero time.Time which is before `oldestAllowed` and therefore even when changing order it will ensure the retention